src/Npgsql/ConnectorPool.cs

```diff
@@ -470,7 +427,7 @@ internal void Release(NpgsqlConnector connector)
                 // too much interlocked operations "contention" at the beginning.
                 var start = Thread.CurrentThread.ManagedThreadId % _max;
 
-                sw = new SpinWait();
+                var sw = new SpinWait();
                 while (true)
```

Comment:
@roji We should be able to cut away this contention on release by doing the following:
1. During AllocateLong where we open a connector and find an open spot in the pool, after we found it, store the array index on the connector.
2. On Release we can take that and do a direct store on that index into the pool.

Added bonus would be that connectors have a fixed position in the array, which should help for a super simple pruning implementation.

For Release() this would only leave:
1. Some contention on Dequeuing on the Release waiters path.
2. 'Unbounded' contention on no waiters, Idle++, Busy-- path.